src/store/actions.js

```diff
@@ -94,7 +95,13 @@ const findIndividualMailboxes = curry((getMailboxes, specialRole) =>
 	)
 )
 
-const combineEnvelopeLists = pipe(flatten, orderBy(prop('dateInt'), 'desc'))
+const combineEnvelopeLists = (sortOrder) => {
+	if (sortOrder === 'oldest-first') {
+		return pipe(flatten, orderBy(prop('dateInt'), 'desc'))
```

Comment:
This looks wrong but it seems to work locally with desc :man_shrugging:

Reply:
This is wrong and caused https://github.com/nextcloud/mail/pull/5495#issuecomment-929907448. After I changed to asc that quirk goes away. Thanks :hugs: